compiler/qsc_passes/src/test_attribute.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_hir::{hir::Attr, visit::Visitor};
+use thiserror::Error;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum TestAttributeError {
+    #[error("This callable has parameters. Tests cannot have parameters.")]
```

Comment:
nitpick: I think the phrasing could be updated to match the style of our other errors, so be one sentence declaring what the problem is, sort of like how the measurement attribute errors look. So something like, "a callable with the test attribute cannot have parameters"